src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/CreateClaimCallbackHandler.java

```diff
@@ -74,6 +87,34 @@ private CallbackResponse validateDateOfBirth(CallbackParams callbackParams) {
             .build();
     }
 
+    private CallbackResponse calculateFee(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        Optional<SolicitorReferences> references = ofNullable(caseData.getSolicitorReferences());
+        String paymentReference = ofNullable(caseData.getPaymentReference())
+            .orElse(references.map(SolicitorReferences::getApplicantSolicitor1Reference).orElse(""));
+
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        List<String> pbaNumbers = getPbaAccounts(authToken);
+
+        CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder()
+            .claimFee(feesService.getFeeDataByClaimValue(caseData.getClaimValue()))
+            .applicantSolicitor1PbaAccounts(DynamicList.fromList(pbaNumbers))
```

Comment:
We should probably check if there's some value selected, otherwise it will be removed when going to previous page and visiting this one again. Might be useful thing to have in `DynamicList` class.

Reply:
I'm not sure how to get this working, the selected value is gone when I click back, maybe one to look at in tech debt?

Reply:
if it's not simple to fix then yeah, we could get that in separate ticket

Reply:
https://tools.hmcts.net/jira/browse/CMC-1122